src/shared/workspace_access_validator.py

```
@@ -186,96 +186,66 @@ def validate_workspace_operation(self, operation: str, workspace_info: Dict[str,
+    def _validate_resource_access(self, resource: Dict[str, Any], workspace_info: Dict[str, Any], user_id: str) -> bool:
+        """Valida acceso a un recurso según el workspace actual"""
+        try:
+            workspace_type = workspace_info.get('workspace_type')
+            workspace_id = workspace_info.get('workspace_id')
+            institute_id = workspace_info.get('institute_id')
+            
+            if workspace_type == 'INSTITUTE':
+                # En workspaces institucionales, validar por institute_id si existe
+                if resource.get('institute_id') and str(resource.get('institute_id')) != str(institute_id):
+                    return False
+                return True
```

Ruff (0.12.2)

Return the negated condition directly

Inline condition

(SIM103)